tests/mocks/MockFileSystem.hh

```diff
@@ -65,6 +65,14 @@ public:
     }
 
     std::unique_ptr<TextWriter> CreateTextFile(const std::wstring& sPath) const override
+    {
+        std::string& sContents = m_mFileContents[sPath];
+        sContents.clear();
+        auto pWriter = std::make_unique<ra::services::impl::StringTextWriter>(sContents);
+        return std::unique_ptr<TextWriter>(pWriter.release());
+    }
+
+    std::unique_ptr<TextWriter> AppendTextFile(const std::wstring& sPath) const override
     {
         std::string& sContents = m_mFileContents[sPath];
```

Comment:
I still think you should use `std::unordered_map::find` for security if `sPath `may not exist, but don't worry about it I can get to it later.

Reply:
i've changed this to use `insert_or_assign`